src/commands/deployCommands/clean.js

```diff
@@ -0,0 +1,79 @@
+const apiMiddleware = require('../deployUtils/apiMiddleware');
+
+const stopOldVersions = (serving, api, versions) =>
+	api
+		.allocations()
+		.then(allocs => {
+			const runningVersions = versions.filter(
+				v => v.servingStatus === 'SERVING'
+			);
+			const servingIds = Object.keys(allocs);
+			console.log('Currently serving traffic:', servingIds.join(', '));
+			console.log(
+				'Currently available to serve:',
+				runningVersions.map(({ id }) => id).join(', ')
+			);
+			const oldestAllocIndex = runningVersions.reduce(
+				(oldestI, v, i) => (servingIds.includes(v.id) ? i : oldestI),
+				-1
+			);
+
+			const stopIndex = Math.max(oldestAllocIndex + serving, 0);
+			const versionsToStop = runningVersions.slice(stopIndex);
+			console.log(
+				'Stopping version(s):',
+				versionsToStop.map(({ id }) => id).join(', ')
+			);
+			// return the versions that should be stopped
+			return versionsToStop;
+		})
+		.then(api.versions.stop);
+
+const deleteOldestVersions = (available, api, versions) => {
+	const versionsToDelete = versions
+		.slice(Math.max(available, 0))
+		.map(({ id }) => id);
+	console.log(`Deleting version(s): ${versionsToDelete.join(', ')}`);
+	return Promise.all(versionsToDelete.map(api.versions.deleteSafe));
```

Comment:
Should this be checking if the version is serving traffic before executing OR does `deleteSafe` have that built in? Sounds like it from the name :)

Reply:
`deleteSafe` does that 👍